src/hooks/prisma.test.js

```diff
@@ -0,0 +1,191 @@
+        .withModel('User')
+        .withOperation('findFirst')
+        .withQueryArgs(`{"where":{"id":${user.id}}}`)
+        .withResult(`{"id":${user.id},"email":"${email}","name":"John Doe"}`)
+        .warm()
+        .build();
+
+      expect(createUserSpan).toEqual(expectedCreateSpan);
+      expect(findUserSpan).toEqual(expectedFindSpan);
+    });
+
+    test('produces an error span on erroneous operation', async () => {
+      const email = getRandomId();
+
+      const user = await client.user.create({ data: { name: 'John Doe', email } });
+      await expect(client.user.create({ data: { name: 'John Doe', email } })).rejects.toThrowError(
+        'Unique constraint failed on the fields'
+      );
+
+      const spans = SpansContainer.getSpans();
+      expect(spans).toHaveLength(2);
+
+      const [successSpan, errorSpan] = spans;
+
+      const expectedSuccessSpan = new PrismaSpanBuilder()
+        .withId(successSpan.id)
+        .withStarted(successSpan.started)
+        .withEnded(successSpan.ended)
+        .withModel('User')
+        .withOperation('create')
+        .withQueryArgs(`{"data":{"name":"John Doe","email":"${email}"}}`)
+        .withResult(`{"id":${user.id},"email":"${email}","name":"John Doe"}`)
+        .build();
+
+      const expectedErrorSpan = new PrismaSpanBuilder()
+        .withId(errorSpan.id)
+        .withStarted(errorSpan.started)
+        .withEnded(errorSpan.ended)
+        .withModel('User')
+        .withOperation('create')
+        .withQueryArgs(`{"data":{"name":"John Doe","email":"${email}"}}`)
+        .withError(expect.stringContaining('Unique constraint failed on the fields'))
+        .warm()
+        .build();
+
+      expect(successSpan).toEqual(expectedSuccessSpan);
+      expect(errorSpan).toEqual(expectedErrorSpan);
+    });
+
+    test('allows other user-extensions to be run', async () => {
+      let extensionExecuted = false;
+
+      const clientWithUserExtension = client.$extends({
+        query: {
+          $allOperations({ model, operation, args, query }) {
+            extensionExecuted = true;
+            return query(args);
+          },
+        },
+      });
+
+      await clientWithUserExtension.user.findFirst({ where: { id: 1 } });
+
+      expect(extensionExecuted).toBe(true);
+    });
+
+    test('produces a raw query span', async () => {
+      const userId = getRandomId();
+      await client.$queryRaw`SELECT * FROM User WHERE id = ${userId};`;
+
+      const spans = SpansContainer.getSpans();
+      expect(spans).toHaveLength(1);
+
+      const [rawQuerySpan] = spans;
+
+      const expectedRawQuerySpan = new PrismaSpanBuilder()
+        .withId(rawQuerySpan.id)
+        .withStarted(rawQuerySpan.started)
+        .withEnded(rawQuerySpan.ended)
+        .withOperation('$queryRaw')
+        .withQueryArgs(
+          `{"values":["${userId}"],"strings":["SELECT * FROM User WHERE id = ",";"]}`
+        )
+        .withResult(`[]`)
```

Comment:
can you add a test with a very long query / result? (i.e. `select * from users`)
we want to trim long contents

Reply:
Sure thing: 55742ae1027952ce285741cf3f65e3560d113876